File: count_sum.rb
# 8kyu Count of positives / sum of negatives
def count_positives_sum_negatives(lst)
  if !lst || lst.length() == 0
    return []
  else
    positives = []
    negatives = []

    lst.each do |el|
      el > 0 ? positives.push(el) : negatives.push(el)
    end

    positive_count = positives.length() || 0
    negative_sum = negatives.reduce(:+) || 0

    [positive_count, negative_sum]
  end
end

# Second solution

def count_positives_sum_negatives(lst)
  return [] if !lst || lst.empty?
  [lst.count(&:positive?) || 0, lst.select(&:negative?).reduce(:+)|| 0]
end

# Refactor to use ternary operator

def count_positives_sum_negatives(lst)
  !lst || lst.empty? ? [] : [lst.count(&:positive?) || 0, lst.select(&:negative?).reduce(:+)|| 0]
end

File: countSum.js
// 8kyu Count of positives / sum of negatives
function countPositivesSumNegatives(input) {
  if (input === null || input.length === 0) {
    return [];
  } else {
  console.log(input);
    let positives = [];
    let negatives = [];

    for (let i = 0; i < input.length; i++) {
      let currEl = input[i];
      currEl > 0 ? positives.push(currEl) : negatives.push(currEl);
    }

    let positiveCount = positives.length;
    let negativeSum = negatives.reduce((sum, num) => sum + num, 0);

    return [positiveCount, negativeSum];
  }
}

// A couple of small refactors
function countPositivesSumNegatives(input) {
  if (!input || input.length === 0) {
    return [];
  } else {
    let positives =[], negatives = [];

    for (let i = 0; i < input.length; i++) {
      let currEl = input[i];
      currEl > 0 ? positives.push(currEl) : negatives.push(currEl);
    }

    let positiveCount = positives.length;
    let negativeSum = negatives.reduce((sum, num) => sum + num, 0);

    return [positiveCount, negativeSum];
  }
}



countPositivesSumNegatives([1, 2, 3, 4, 5, 6, 7, 8, 9, 10, -11, -12, -13, -14, -15]); //[10, -65]
